homeassistant/components/xiaomi_miio/const.py

```diff
@@ -194,3 +197,94 @@
 SERVICE_CLEAN_SEGMENT = "vacuum_clean_segment"
 SERVICE_CLEAN_ZONE = "vacuum_clean_zone"
 SERVICE_GOTO = "vacuum_goto"
+
+
+# Features
+FEATURE_SET_BUZZER = 1
+FEATURE_SET_LED = 2
+FEATURE_SET_CHILD_LOCK = 4
+FEATURE_SET_LED_BRIGHTNESS = 8
+FEATURE_SET_FAVORITE_LEVEL = 16
+FEATURE_SET_AUTO_DETECT = 32
+FEATURE_SET_LEARN_MODE = 64
+FEATURE_SET_VOLUME = 128
+FEATURE_RESET_FILTER = 256
+FEATURE_SET_EXTRA_FEATURES = 512
+FEATURE_SET_TARGET_HUMIDITY = 1024
+FEATURE_SET_DRY = 2048
+FEATURE_SET_FAN_LEVEL = 4096
+FEATURE_SET_MOTOR_SPEED = 8192
+
+FEATURE_FLAGS_AIRPURIFIER = (
+    FEATURE_SET_BUZZER
+    | FEATURE_SET_CHILD_LOCK
+    | FEATURE_SET_LED
+    | FEATURE_SET_LED_BRIGHTNESS
+    | FEATURE_SET_FAVORITE_LEVEL
+    | FEATURE_SET_LEARN_MODE
+    | FEATURE_RESET_FILTER
+    | FEATURE_SET_EXTRA_FEATURES
+)
```

Comment:
A slightly offtopic comment here, but I thought I'd write it down nevertheless :-)

Considering that the feature flags would also be useful for other users of python-miio, I'd personally like to see them moved to the backend library. If the backend library would be the single source of truth for this information, adding support for new devices at homeassistant's end would be a simple version bump.

I have written previously about some changes I'd like to perform on the backend lib whenever/if I find time for that -- e.g., https://github.com/rytilahti/python-miio/issues/1059#issuecomment-851070398 https://github.com/rytilahti/python-miio/pull/1079#discussion_r658146499 -- with the goal to simplify the upstream/downstream interoperability. The driving idea of all those changes would be to make it simpler for homeassistant (and other downstreams) to interact with the library & reduce the need for code duplication among projects.

Would you be interest to help me to brainstorm/implement/test the necessary changes to the backend lib to make the idea of having feature flags (as a first step toward the direction I described in those links above) happen?

Reply:
Let me now what constants you want to add. For the switch platform I have added FEATURE_SET_CLEAN = 16384.
It is is not in this PR.